src/Service/Parser/UnsupportedTupleCombinationParser.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service\Parser;
+
+use Lmc\ApiFilter\Exception\InvalidArgumentException;
+
+class UnsupportedTupleCombinationParser extends AbstractParser
+{
+    public function supports(string $rawColumn, $rawValue): bool
+    {
+        return $this->isTuple($rawColumn) || $this->isTuple($rawValue);
+    }
+
+    public function parse(string $rawColumn, $rawValue): iterable
+    {
+        throw new InvalidArgumentException(sprintf(
+            'Invalid combination of a tuple and a scalar. Column %s and value %s.',
+            $this->formatForException($rawColumn),
+            $this->formatForException($rawValue)
+        ));
+    }
+
+    private function formatForException($value): string
```

Comment:
Maybe `formatValueForException($inputValue)` to make it a bit more clear?

Reply:
It formats both `column` and `value` so I would see it more misleading..